benchmark/testsystems/bookkeepers.py

```diff
@@ -220,30 +196,49 @@ def sample_v_given_x(self, x):
         self.simulation.context.applyVelocityConstraints(self.constraint_tolerance)
         return get_velocities(self.simulation)
 
-    def accumulate_shadow_work(self, x_0, v_0, n_steps, store_potential_energy=False):
+    def accumulate_shadow_work(self, x_0, v_0, n_steps, box_vectors=None, store_potential_energy=False, store_W_shad_trace=False):
         """Run the integrator for n_steps and return the change in energy - the heat."""
-        get_energy = lambda: get_total_energy(self.simulation)
+        get_energy = lambda: get_total_energy(self.simulation).value_in_unit(W_unit)
```

Comment:
You're mixing retrieval of values in specific units (`W_unit`) with retrieval of unit-bearing quantities in default dimensionless units (`integrator.getGlobalVariableByName("heat")`). While this works now, if someone were to simply change `W_unit`, it would break things. In general, we want to avoid the potential for mistakes to crop up if something seemingly arbitrary is changed.